File: server/php/src/Application/Interfaces/Services/PaymentServiceInterface.php
<?php

namespace OnlinePayments\ExampleApp\Application\Interfaces\Services;

use OnlinePayments\ExampleApp\Application\DTOs\AdditionalPaymentActions\RequestDto;
use OnlinePayments\ExampleApp\Application\DTOs\AdditionalPaymentActions\ResponseDto;
use OnlinePayments\ExampleApp\Application\DTOs\GetPaymentDetails\ResponseDto as GetPaymentDetailsResponseDto;
use OnlinePayments\ExampleApp\Application\DTOs\Payment\RequestDto as CreatePaymentRequestDto;
use OnlinePayments\ExampleApp\Application\DTOs\Payment\ResponseDto as CreatePaymentResponseDto;

interface PaymentServiceInterface
{
    public function createPayment(CreatePaymentRequestDto $requestDto): CreatePaymentResponseDto;
    public function getPaymentDetailsById(string $id): GetPaymentDetailsResponseDto;
    public function capturePayment(RequestDto $requestDto): ResponseDto;
    public function refundPayment(RequestDto $requestDto): ResponseDto;
    public function cancelPayment(RequestDto $requestDto): ResponseDto;
}

File: server/python/app/application/interfaces/services/payment_service_interface.py
from abc import ABC, abstractmethod

from app.application.dtos.additional_payment_actions.response_dto import AdditionalPaymentActionsResponseDto
from app.application.dtos.get_payment_details.response_dto import GetPaymentDetailsResponseDto
from app.application.dtos.create_payment.request_dto import CreatePaymentRequestDto
from app.application.dtos.create_payment.response_dto import CreatePaymentResponseDto

class PaymentServiceInterface(ABC):

    @abstractmethod
    def create_payment(self, request_dto: CreatePaymentRequestDto) -> CreatePaymentResponseDto:
        pass

    @abstractmethod
    def capture_payment(self, request_dto) -> AdditionalPaymentActionsResponseDto:
        pass

    @abstractmethod
    def refund_payment(self, request_dto) -> AdditionalPaymentActionsResponseDto:
        pass

    @abstractmethod
    def cancel_payment(self, request_dto) -> AdditionalPaymentActionsResponseDto:
        pass

    @abstractmethod
    def get_payment_details(self, id: str) -> GetPaymentDetailsResponseDto:
        pass